utils/profile.ts

```diff
@@ -235,52 +242,82 @@ Task<IGetLimitedProfileResponses> =>
         );
       })
     ),
-    TE.chainW(profile => {
-      // To determine allowance, use a different algorithm depending in subscription mode
-      const isSenderAllowedTask =
-        profile.servicePreferencesSettings.mode ===
-        ServicesPreferencesModeEnum.LEGACY
-          ? isSenderAllowedLegacy(
-              profile.blockedInboxOrChannels,
-              userAttributes.service.serviceId
-            )
-          : isSenderAllowed(
-              servicesPreferencesModel,
-              userAttributes.service.serviceId,
-              profile.fiscalCode,
-              profile.servicePreferencesSettings
-            );
-
-      return pipe(
-        isSenderAllowedTask,
-        TE.bimap(
-          error =>
-            error.kind === "UNEXPECTED_VALUE"
-              ? ResponseErrorInternal(`Unexpected mode: ${error.value}`)
-              : ResponseErrorQuery(
-                  "Failed to read preference for the given service",
-                  error
-                ),
-          isAllowed => ({
-            isAllowed,
-            profile
-          })
+    TE.chain(profile =>
+      pipe(
+        O.fromNullable(userAttributes.service.serviceMetadata),
+        O.map(serviceMetadata => serviceMetadata.category),
+        TE.fromPredicate(
+          category =>
+            O.isSome(category) &&
+            category.value === SpecialServiceCategoryEnum.SPECIAL,
+          identity
```

Comment:
```suggestion
               profile,
        TE.fromPredicate(
          () =>
            userAttributes.service.serviceMetadata?.category ===
            SpecialServiceCategoryEnum.SPECIAL,
          identity
```
Could we simplify it with something like this?

Reply:
I tried it yesterday on another PR too, but I ended thinking that passing from an Option, like @BurnedMarshal did, gives a more precise idea of what we want to obtain. The second one, even if more simple and concise is not declaring an intent imho... it seems more "imperative" to me. But I think it's @BurnedMarshal choice...